internal/gengapic/gengapic.go

```diff
@@ -149,30 +230,30 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 		// Keep the current behavior for now, but we could revisit this later.
 		outFile := pbinfo.ReduceServName(s.GetName(), "")
 		outFile = camelToSnake(outFile)
-		outFile = filepath.Join(outDir, outFile)
+		outFile = filepath.Join(g.opts.outDir, outFile)
 
 		g.reset()
-		if err := g.gen(s, pkgName); err != nil {
+		if err := g.gen(s, g.opts.pkgName); err != nil {
```

Comment:
Here and below, do we want to pass `g.opts.pkgName` to `g.gen()`, or have the method look at its `g.opts` directly? The latter might be simpler while not hindering testing via fake values, etc.

Reply:
Good point. My original intention was to limit the blast radius of changed code, but it's better and more consistent to just use the attribute.